indico/modules/events/registration/fields/simple.py

```diff
@@ -563,38 +564,60 @@ def _format_item(uuid, number_of_slots):
         return sorted(_format_item(uuid, number_of_slots) for uuid, number_of_slots in reg_data.iteritems())
 
     def process_form_data(self, registration, value, old_data=None, billable_items_locked=False):
-        # TODO: create new data version here if an item from old_data's version
-        # was already chosen but doesn't exist anymore (or has a new price) and
-        # the user also selected a new item that doesn't exist in the current
-        # version yet.
 
         # always store no-option as empty dict
         if value is None:
             value = {}
+
+        return_value = {}
+
+        if old_data:
+            old_choices_mapping = {x['id']: x for x in old_data.field_data.versioned_data['choices']}
+            new_choices_mapping = {x['id']: x for x in self.form_item.versioned_data['choices']}
+            revoked_selected_choice_ids = {}
```

Comment:
those two assignments are now unused (a decent editor/ide highlights this :p)

![ss](https://cloud.githubusercontent.com/assets/179599/11034436/e717039a-86eb-11e5-8794-bdc4f257f821.png)